### ISSUE TITLE
URL copied to clipboard via share extension does not match location in URL bar

### ISSUE THREAD

**Issue by codeactual** [brave/brave-ios#5238]
### Description: 

When sharing the active tab's URL to the clipboard, the copied URL does not match the one displayed at the top of the screen.

### Steps to Reproduce 
  1. Open https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address
  2. Tap `...` at the bottom-right of the screen.
  3. Tap `Share with...`
  4. Tap `Copy`

**Actual result:** 

This is copied to the clipboard: http://ghdocs-prod.azurewebsites.net/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address

**Expected result:**

The same URL shown at the top of the screen: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

Easily reproduced

**Brave Version:** 

1.36 (22.3.11.16)

**Device details:** 

iPhone 11 (iOS 15.4.1)

**Website problems only:**
- did you check with Brave Shields down? Yes, no change.
- did you check in Safari/Firefox (WkWebView-based browsers)? Yes, Safari copies the page title as a link and the URL is the expected one above.

### Additional Information 

The unexpected URL exists in twice in the document (but with an explicit `:80`), once in an `<meta property="og:url">` and another in a `<script>` near the end.

**Comment by kylehickinson** [comment]
Some minor investigation: It seems on mobile GH Docs pages it includes this meta property which is where the url comes from, will have to dig deeper to understand why its being used as a share url

```html
<meta property="og:url" content="http://ghdocs-prod.azurewebsites.net:80/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address">
```

**Comment by iccub** [comment]
Seems to work on 1.37(checked on `development` on simulator), you can try to update and see @codeactual, the build is live in App Store

In 1.37 we've made few improvements to url detection

**Comment by codeactual** [comment]
It's fixed for me in 1.37. Thank you.
